### PR TITLE
CLC-5137, db record if Webcast tool made visible in Canvas course site

### PR DESCRIPTION
**Pull request by johncrossman**
https://jira.ets.berkeley.edu/jira/browse/CLC-5137

For the moment, the webcast:canvas_integration rake task takes a single canvas_course_site_id and updates db if the webcast tool is 'unhidden' in course site.  Much of the wiring is not done but this is a step in the right direction.

The job will need to know Canvas app_id of webcast tool so that config has been added to yml. Release instructions will get update if this PR is merged.

### CODE REVIEW

**Comment by johncrossman** [comment]
FWIW, Bamboo build passed: https://bamboo.ets.berkeley.edu/bamboo/browse/MYB-CJT51-1

File: config/settings.yml [hunk 1]
```diff
@@ -71,6 +71,7 @@ canvas_proxy:
   projects_term_id: '5494'
   projects_owner_role_id: '1766'
   default_grading_scheme_id: '4068930'
+  webcast_app_id: '32692'
```
Comment (raydavis) [inline config/settings.yml]: It's brittle to manually configure these, @johncrossman . Usually the way we find the magic tool ID is by looking for a tool match in Canvas's ``external_tools`` API. For example code, see:

https://github.com/ets-berkeley-edu/calcentral/blob/master/app/models/canvas/reconfigure_external_apps.rb
https://github.com/ets-berkeley-edu/calcentral/blob/master/app/models/canvas/external_tools.rb

Is there any way you can make that approach work for this task?

Comment (johncrossman) [inline config/settings.yml]: Thanks @raydavis  I'll have a look. I assume I can make it work.

**Comment by johncrossman** [comment]
New usage: 
```
rake canvas:webcast_lti_refresh course_id=[course_id]
```
And per @raydavis comment I use Canvas::ExternalTools to grab webcast tool id.

File: app/models/canvas/webcast_lti_refresh.rb [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+            unhidden_date = record.webcast_tool_unhidden_at.strftime('%m/%d/%Y')
+            Rails.logger.warn "Do nothing to course site #{@course_id} because Webcast tool was unhidden on #{unhidden_date}."
+          else
+            # TODO: Use Canvas Tabs API to un-hide Webcast tool @canvas_webcast_tool_id
+            @course_site_log_entry = Webcast::CourseSiteLog.create({ canvas_course_site_id: @course_id, webcast_tool_unhidden_at: Time.zone.now })
+            Rails.logger.warn "The Webcast tool has been un-hidden on course site #{@course_id}"
+          end
+        end
+      end
+      @course_site_log_entry
```
Comment (paulkerschen) [inline app/models/canvas/webcast_lti_refresh.rb]: @johncrossman Minor, but any reason this is an instance variable? It doesn't seem like it needs to persist past the method call.

Comment (johncrossman) [inline app/models/canvas/webcast_lti_refresh.rb]: No good reason, @raydavis   Fixed.

**Comment by paulkerschen** [comment]
Looks good!

**Pull request merged**